packages/flutter/lib/src/scheduler/scheduler.dart

```diff
@@ -0,0 +1,151 @@
+
+  int get value => _value;
+  final int _value;
+
+  /// Returns a priority relative to this priority.
+  ///
+  /// A positive [offset] indicates a higher priority.
+  ///
+  /// The parameter [offset] is clamped to +/-[kMaxOffset].
+  Priority operator +(int offset) {
+    if (offset.abs() > kMaxOffset) {
+      // Clamp the input offset.
+      offset = kMaxOffset * offset.sign;
+    }
+    return new Priority._(_value + offset);
+  }
+
+  /// Returns a priority relative to this priority.
+  ///
+  /// A positive offset indicates a lower priority.
+  ///
+  /// The parameter [offset] is clamped to +/-[kMaxOffset].
+  Priority operator -(int offset) => this + (-offset);
+}
+
+/// Scheduler running tasks with specific priorities.
+///
+/// Combines the task's priority with remaining time in a frame to decide when
+/// the task should be run.
+///
+/// Tasks always run in the idle time after a frame has been committed.
+class TaskScheduler {
+  TaskScheduler._();
+
+  SchedulingStrategy schedulingStrategy = new DefaultSchedulingStrategy();
+
+  final PriorityQueue _queue = new HeapPriorityQueue<_SchedulerEntry>(
+    (_SchedulerEntry e1, _SchedulerEntry e2) {
+      // Note that we inverse the priority.
+      return -e1.priority.compareTo(e2.priority);
+    }
+  );
+
+  /// Wether this scheduler already requested to be woken up as soon as
+  /// possible.
+  bool _wakingNow = false;
+
+  /// Wether this scheduler already requested to be woken up in the next frame.
+  bool _wakingNextFrame = false;
+
+  /// Schedules the given [task] with the given [priority].
+  void schedule(VoidCallback task, Priority priority) {
+    bool isFirstTask = _queue.isEmpty;
+    _queue.add(new _SchedulerEntry(task, priority._value));
+    if (isFirstTask) _wakeNow();
+  }
+
+  /// Invoked by the system when there is time to run tasks.
+  void tick() {
+    if (_queue.isEmpty)
+      return;
+    _SchedulerEntry entry = _queue.first;
+    if (schedulingStrategy.shouldRunTaskWithPriority(entry.priority)) {
+      try {
+        (_queue.removeFirst().task)();
+      } finally {
+        if (_queue.isNotEmpty)
+          _wakeNow();
+      }
+    } else {
+      _wakeNextFrame();
+    }
+  }
+
+  /// Tells the system that the scheduler is awake and should be called as
+  /// soon a there is time.
+  void _wakeNow() {
+    if (_wakingNow)
+      return;
+    _wakingNow = true;
+    Timer.run(() {
+      _wakingNow = false;
+      tick();
+    });
+  }
+
+  /// Tells the system that the scheduler needs to run again (ideally next
+  /// frame).
+  void _wakeNextFrame() {
+    if (_wakingNextFrame)
+      return;
+    _wakingNextFrame = true;
+    animation.scheduler.requestAnimationFrame((_) {
+      _wakingNextFrame = false;
+      // RequestAnimationFrame calls back at the beginning of a frame. We want
+      // to run in the idle-phase of an animation. We therefore request to be
+      // woken up as soon as possible.
+      _wakeNow();
```

Comment:
We can also elaborate the other scheduler.dart.  My point is just that going through the event loop here isn't what you want.  You want to run at defined time at the start of the idle period.  What you've got here is fine for the first iteration, but we'll eventually want to merge the two scheduling systems, and when we do, we'll probably change the timing of this callback.